### PR TITLE
Update to Cake 2.0

### PR DESCRIPTION
**Pull request by twenzel**
- Update to use Cake 2.0
- Support Cake 2.0 as well as .NET 6
- Update to use VS 2022 build image

Fixes #78, #77, #79, #76

### CODE REVIEW

File: src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj [hunk 1]
```diff
@@ -1,26 +1,23 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net50;netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net461;netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
```
Comment (nils-a) [inline src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj:3]:
```suggestion
    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
```

`net461` is no longer a requirement, as of https://cakebuild.net/docs/extending/addins/best-practices#net-target-version

File: src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
   <PropertyGroup>
-    <TargetFrameworks>net50;netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net461;netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
   </PropertyGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net461' ">
+    <PackageReference Include="Cake.Testing" Version="1.0.0"  />
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' != 'net461' ">
+    <PackageReference Include="Cake.Testing" Version="2.0.0" />
+  </ItemGroup>
```
Comment (nils-a) [inline src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj:10]:
```suggestion
  <ItemGroup>
    <PackageReference Include="Cake.Testing" Version="2.0.0" />
  </ItemGroup>
```

Those are not needed, as `net461` is no longer needed.

File: src/Cake.FileHelpers/Cake.FileHelpers.csproj [hunk 3]
```diff
@@ -1,7 +1,6 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net50;netstandard2.0;net46</TargetFrameworks>
-    <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netstandard1.6' ">1.6.1</NetStandardImplicitPackageVersion>
+    <TargetFrameworks>netstandard2.0;net461;netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
```
Comment (nils-a) [inline src/Cake.FileHelpers/Cake.FileHelpers.csproj:3]:
```suggestion
    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
```

Both, `net461` and `netstandard2.0` are no longer a requirement, as of https://cakebuild.net/docs/extending/addins/best-practices#net-target-version

File: src/Cake.FileHelpers/Cake.FileHelpers.csproj [hunk 4]
```diff
@@ -22,13 +21,17 @@
...
     <PackageReference Include="Cake.Core" Version="1.0.0" PrivateAssets="All" />
     <PackageReference Include="Cake.Common" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="CakeContrib.Guidelines" Version="1.2.0">
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' != 'netstandard2.0' And '$(TargetFramework)' != 'net461' ">
+    <PackageReference Include="Cake.Common" Version="2.0.0" PrivateAssets="All" />
+    <PackageReference Include="Cake.Core" Version="2.0.0" PrivateAssets="All" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="CakeContrib.Guidelines" Version="1.3.0">
```
Comment (nils-a) [inline src/Cake.FileHelpers/Cake.FileHelpers.csproj:34]: Those conditions can also be removed, as `net461` and `netstandard2.0` are no longer needed.

**Review by nils-a — CHANGES_REQUESTED** [review]
Thank you for the PR.
You can, however, drop the `net461` and `netstandard2.0` TFMs. Cake 2.0 dropped support for those.

**Review by augustoproiete — CHANGES_REQUESTED** [review]
Looks good. Same comments as @nils-a + .NET 6 SDK install needed

File: src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj (hunk 1 above)
Comment (twenzel) [reply src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj]: done

File: src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj (hunk 2 above)
Comment (twenzel) [reply src/Cake.FileHelpers.Tests/Cake.FileHelpers.Tests.csproj]: done

File: src/Cake.FileHelpers/Cake.FileHelpers.csproj (hunk 3 above)
Comment (twenzel) [reply src/Cake.FileHelpers/Cake.FileHelpers.csproj]: done

File: src/Cake.FileHelpers/Cake.FileHelpers.csproj (hunk 4 above)
Comment (twenzel) [reply src/Cake.FileHelpers/Cake.FileHelpers.csproj]: done

**Comment by twenzel** [comment]
.NET 6 SDK is already installed because appveyor uses the `Visual Studio 2022` image.

**Comment by twenzel** [comment]
@augustoproiete  Should I update the Github 'Build' action and use the `setup-dotnet` task to ensure .NET 6 SDK is installed?

**Comment by augustoproiete** [comment]
> .NET 6 SDK is already installed because appveyor uses the `Visual Studio 2022` image.

You're correct, except that this project no longer seems to be using AppVeyor for builds... Instead builds are using GitHub actions, so .NET 6 install should be here: https://github.com/cake-contrib/Cake.FileHelpers/blob/4.0.1/.github/workflows/build.yml

Example from @nils-a https://github.com/cake-contrib/Cake.ExtendedNuGet/blob/5.0.0/.github/workflows/build.yml#L31-L38

**Comment by augustoproiete** [comment]
> @augustoproiete Should I update the Github 'Build' action and use the `setup-dotnet` task to ensure .NET 6 SDK is installed?

Yes, please 😄

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -46,6 +46,11 @@ jobs:
 
       - name: Fetch all tags and branches
         run: git fetch --prune --unshallow
+        
+      - name: Install .NET SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.x'
```
Comment (augustoproiete) [inline .github/workflows/build.yml:53]: We aim for [deterministic builds](https://cakebuild.net/docs/writing-builds/reproducible-builds/), so we should add the exact `major.minor.patch` for all 3 SDKs we target, so there are no surprises

**Review by augustoproiete — CHANGES_REQUESTED** [review]

**Review by augustoproiete — DISMISSED** [review]
LGTM

**Review by nils-a — DISMISSED** [review]

**Review by nils-a — APPROVED** [review]

**Comment by nils-a** [comment]
@twenzel your changes have been merged, thanks for your contribution :+1:

**Pull request merged**